main/main.ino

```diff
@@ -1037,106 +1049,105 @@ void loop() {
 
 #if defined(ESP8266) || defined(ESP32)
   if (WiFi.status() == WL_CONNECTED) {
+    ArduinoOTA.handle();
```

Comment:
is this blocking?

Reply:
I have checked and it is not blocking.

Reply:
according to https://github.com/esp8266/Arduino/blob/master/libraries/ArduinoOTA/ArduinoOTA.cpp I don't see why it should not block, It runs on the same thread as the loop so it should be blocking.

Reply:
In normal conditions ...

Reply:
In which case it does not block the loop thread while doing an update. And on which thread the update is then executed?

Reply:
Maybe the ESP32 does not use the same implementation, but from my test I confirm it is not blocking:
```
void loop() {
Serial.println("begin times");
times = millis();
Serial.println(times);
  ArduinoOTA.handle();
Serial.println("end times");
times = millis();
Serial.println(times);
}
```

I was able to see the serial monitors prints when uploading with OTA in parallel.

Reply:
then we should make the OTA update blocking by waiting until it's finished.

Reply:
Do you have some proposal so as to do that ? Or do I just add the return when we detect that the OTA is in progress through the ProcessLock status?

The state of the OTA is not a public property from what I have seen.

Reply:
https://github.com/espressif/arduino-esp32/blob/master/libraries/ArduinoOTA/src/ArduinoOTA.cpp this is the implementation for the ESP32, it's a little bit different form the esp8266 one but also blocking (or I don't see where the execution is moved to another thread).

Reply:
I'm redoing the test to confirm or contradict

Reply:
It's strange, I have 2 different behaviours:
* Serial monitor started before the OTA upload, the loop continue displaying prints while the OTA is in progress,
* Serial monitor started during the OTA upload, I see only OTA progress and not the loop prints.

So I have modified my sketch to display into the` .onProgress `callback a variable incremented in the loop, and guess what, it is blocking!
```
1392137s: 0%
1392137s: 0%
1392137s: 0%
1392137s: 0%
1392137s: 0%
1392137s: 0%
1392137s: 0%
1392137s: 0%
1392137s: 1%
1392137s: 1%
1392137s: 1%
1392137s: 1%
1392137s: 1%
1392137s: 1%
1392137s: 1%
1392137s: 2%
1392137s: 2%
1392137s: 2%
1392137s: 2%
1392137s: 2%
1392137s: 2%
1392137s: 2%
1392137s: 3%
1392137s: 3%
1392137s: 3%
1392137s: 3%
1392137s: 3%
1392137s: 3%
1392137s: 3%
1392137s: 3%
1392137s: 4%
1392137s: 4%
1392137s: 4%
1392137s: 4%
1392137s: 4%
1392137s: 4%
1392137s: 4%
1392137s: 5%
1392137s: 5%
1392137s: 5%
1392137s: 5%
1392137s: 5%
1392137s: 5%
1392137s: 5%
1392137s: 6%
1392137s: 6%
1392137s: 6%
1392137s: 6%
1392137s: 6%
1392137s: 6%
1392137s: 6%
1392137s: 7%
1392137s: 7%
1392137s: 7%
1392137s: 7%
1392137s: 7%
1392137s: 7%
1392137s: 7%
1392137s: 7%
1392137s: 8%
1392137s: 8%
1392137s: 8%
1392137s: 8%
1392137s: 8%
1392137s: 8%
1392137s: 8%
1392137s: 9%
1392137s: 9%
1392137s: 9%
1392137s: 9%
1392137s: 9%
1392137s: 9%
1392137s: 9%
1392137s: 10%
1392137s: 10%
1392137s: 10%
1392137s: 10%
1392137s: 10%
1392137s: 10%
1392137s: 10%
1392137s: 10%
1392137s: 11%
1392137s: 11%
1392137s: 11%
1392137s: 11%
1392137s: 11%
1392137s: 11%
1392137s: 11%
1392137s: 12%
1392137s: 12%
1392137s: 12%
1392137s: 12%
1392137s: 12%
1392137s: 12%
1392137s: 12%
1392137s: 13%
1392137s: 13%
1392137s: 13%
1392137s: 13%
1392137s: 13%
1392137s: 13%
1392137s: 13%
1392137s: 14%
1392137s: 14%
1392137s: 14%
1392137s: 14%
1392137s: 14%
1392137s: 14%
1392137s: 14%
1392137s: 14%
1392137s: 15%
1392137s: 15%
1392137s: 15%
1392137s: 15%
```
Sorry for the bad result of the first test.

I will remove the checks that are not related to second core use.